app/models/miq_alert.rb

```diff
@@ -213,7 +213,7 @@ def evaluate(target, inputs = {})
   end
 
   def add_status_post_evaluate(target, result)
-    existing = miq_alert_statuses.find_by(:resource_type => target.class.base_class.name, :resource_id => target.id)
+    existing = miq_alert_statuses.find_by(:resource => target)
```

Comment:
Happened to spot another in this file: https://github.com/durandom/manageiq/blob/dont_use_sti_type/app/models/miq_alert.rb#L184